### PR TITLE
prototype output

### PR DESCRIPTION
**Pull request by vercibar**

### CODE REVIEW

File: uml/class_diagrams/class.plantuml [hunk 1]
```diff
@@ -227,6 +245,10 @@ TeleportGate .up.|> INeighbour
 RobotRecipe .up.|> IRecipe
 TeleporterRecipe .up.|> IRecipe
 Settler .up.|> IStorageCapable
+AbstractCrewmate ...|> IIdentifiable
```
Comment (triszt4n) [inline uml/class_diagrams/class.plantuml:248]: nem az AbstractCrewmate, hanem AbstractSpaceship kén ide

**Review by triszt4n — CHANGES_REQUESTED** [review]

File: uml/class_diagrams/class.plantuml (hunk 1 above)
Comment (triszt4n) [reply uml/class_diagrams/class.plantuml:248]: Illetve a SunController kimaradt

Comment (vercibar) [reply uml/class_diagrams/class.plantuml:248]: I am a fool.

**Review by triszt4n — APPROVED** [review]
# 🛴

File: java/src/hu/holyoil/resource/Coal.java [hunk 2]
```diff
@@ -13,6 +14,15 @@
...
     public Coal() {
+        id = Main.GetId();
+    }
+
+    /**
+     * Kiírja a szenet emberileg olvasható formátumban.
+     * */
+    @Override
+    public String toString() {
+        return "COAL: " + id;
```
Comment (triszt4n) [inline java/src/hu/holyoil/resource/Coal.java:25]: Do we need the `:`?

File: java/src/hu/holyoil/resource/Iron.java [hunk 3]
```diff
@@ -11,7 +12,15 @@
...
     public Iron() {
+        id = Main.GetId();
+    }
 
+    /**
+     * Kiírja a vasat emberileg olvasható formátumban.
+     * */
+    @Override
+    public String toString() {
+        return "IRON: " + id;
```
Comment (triszt4n) [inline java/src/hu/holyoil/resource/Iron.java:23]: ?

File: java/src/hu/holyoil/resource/Uranium.java [hunk 4]
```diff
@@ -15,12 +16,20 @@
...
         health = 3;
+        id = Main.GetId();
+    }
+
+    /**
+     * Kiírja az uránt emberileg olvasható formátumban.
+     * */
+    @Override
+    public String toString() {
+        return "URANIUM: " + id + " " + health;
```
Comment (triszt4n) [inline java/src/hu/holyoil/resource/Uranium.java:32]: ?

File: java/src/hu/holyoil/resource/Water.java [hunk 5]
```diff
@@ -13,7 +14,17 @@
...
     public Water() {
+        id = Main.GetId();
     }
+
+    /**
+     * Kiírja a vizet emberileg olvasható formátumban.
+     * */
+    @Override
+    public String toString() {
+        return "WATER: " + id;
```
Comment (triszt4n) [inline java/src/hu/holyoil/resource/Water.java:25]: ?

**Pull request merged**